src/stan/mcmc/hmc/nuts_classic/adapt_diag_e_nuts_classic.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef STAN_MCMC_HMC_NUTS_ADAPT_DIAG_E_NUTS_CLASSIC_HPP
+#define STAN_MCMC_HMC_NUTS_ADAPT_DIAG_E_NUTS_CLASSIC_HPP
+
+#include <stan/interface_callbacks/writer/base_writer.hpp>
+#include <stan/mcmc/stepsize_var_adapter.hpp>
+#include <stan/mcmc/hmc/nuts_classic/diag_e_nuts_classic.hpp>
+
+namespace stan {
+  namespace mcmc {
+
+    // The No-U-Turn Sampler (NUTS) on a
+    // Euclidean manifold with diagonal metric
+    // and adaptive stepsize
+
+    template <class Model, class BaseRNG>
+    class adapt_diag_e_nuts_classic:
+      public diag_e_nuts_classic<Model, BaseRNG>,
```

Comment:
Multiple inheritance is a really dangerous pattern.  I'm not saying you have to change this here, but you really want to think about including a reference to a class rather than using multiple inheritance.  It's much easier to reason about and usually the relationship you have is not an is-a relationship.  Here it'd seem to make sense to have the `stepsize_var_adaptor` be held as a member variable.